rpc/erigon.go

```diff
@@ -768,3 +770,339 @@ type Eth1InternalTransactionWithPosition struct {
 	types.Eth1InternalTransaction
 	txPosition int
 }
+
+type BlockResponse struct {
+	Hash          string                    `json:"hash"`
+	ParentHash    string                    `json:"parentHash"`
+	UncleHash     string                    `json:"uncleHash"`
+	Coinbase      string                    `json:"coinbase"`
+	Root          string                    `json:"stateRoot"`
+	TxHash        string                    `json:"transactionsHash"`
+	ReceiptHash   string                    `json:"receiptsHash"`
+	Difficulty    string                    `json:"difficulty"`
+	Number        string                    `json:"number"`
+	GasLimit      string                    `json:"gasLimit"`
+	GasUsed       string                    `json:"gasUsed"`
+	Time          string                    `json:"timestamp"`
+	Extra         string                    `json:"extraData"`
+	MixDigest     string                    `json:"mixHash"`
+	Bloom         string                    `json:"logsBloom"`
+	Uncles        []*geth_types.Block       `json:"uncles"`
+	Transactions  []*geth_types.Transaction `json:"transactions"`
+	Withdrawals   []*geth_types.Withdrawal  `json:"withdrawals"`
+	BlobGasUsed   *string                   `json:"blobGasUsed"`
+	ExcessBlobGas *string                   `json:"excessBlobGas"`
+	BaseFee       string                    `json:"baseFee"`
+}
+
+func (b *BlockResponse) UnmarshalJSON(data []byte) error {
```

Comment:
Have a look at https://github.com/ethereum/go-ethereum/blob/master/ethclient/ethclient.go#L128
you could reuse some of the code done there for uncles check and header parsing